File: lab07/lab07_example00/scripts/lab07_example00_write.py
#!/usr/bin/env python
import rospy
import rosbag
from std_msgs.msg import Int32, String
import rospkg

def lab07_example00_write():

    rospy.init_node('lab07_example00_write')

    rospack = rospkg.RosPack()

    path = rospack.get_path('lab07_example00')
    bag = rosbag.Bag(path + '/../example00_data.bag', 'w')

    try:
		str = String()
		str.data = 'foo'

		i = Int32()
		i.data = 42

		bag.write('chatter', str)
		bag.write('numbers', i)
    finally:
        bag.close()

if __name__ == '__main__':
    try:
        lab07_example00_write()
    except rospy.ROSInterruptException:
        pass

File: lab07/lab07_example00/src/lab07_example00_write.cpp
#include "ros/ros.h"
#include <rosbag/bag.h>
#include <std_msgs/Int32.h>
#include <std_msgs/String.h>


int main(int argc, char **argv)
{

    ros::init(argc, argv, "lab7example00_write");

    rosbag::Bag bag;
    bag.open(MY_BAG_PATH, rosbag::bagmode::Write);

    ros::NodeHandle nh;

    std_msgs::String str;
    str.data = std::string("foo");

    std_msgs::Int32 i;
    i.data = 42;

    bag.write("chatter", ros::Time::now(), str);
    bag.write("numbers", ros::Time::now(), i);


    ros::spinOnce();

    bag.close();
    return 5;
}